src/DotNetWorker.Grpc/GrpcWorker.cs

```diff
@@ -78,54 +67,16 @@ internal class GrpcWorker : IWorker
 
         public async Task StartAsync(CancellationToken token)
         {
-            var eventStream = _rpcClient.EventStream(cancellationToken: token);
-
-            await SendStartStreamMessageAsync(eventStream.RequestStream);
-
-            _ = StartWriterAsync(eventStream.RequestStream);
-            _ = StartReaderAsync(eventStream.ResponseStream);
+            _workerClient = await _workerClientFactory.StartClientAsync(this, token);
         }
 
-        public Task StopAsync(CancellationToken token)
-        {
-            return Task.CompletedTask;
-        }
+        public Task StopAsync(CancellationToken token) => Task.CompletedTask;
 
-        private async Task SendStartStreamMessageAsync(IClientStreamWriter<StreamingMessage> requestStream)
-        {
-            StartStream str = new StartStream()
-            {
-                WorkerId = _startupOptions.WorkerId
-            };
-
-            StreamingMessage startStream = new StreamingMessage()
-            {
-                StartStream = str
-            };
-
-            await requestStream.WriteAsync(startStream);
-        }
-
-        private async Task StartWriterAsync(IClientStreamWriter<StreamingMessage> requestStream)
-        {
-            await foreach (StreamingMessage rpcWriteMsg in _outputReader.ReadAllAsync())
-            {
-                await requestStream.WriteAsync(rpcWriteMsg);
-            }
-        }
-
-        private async Task StartReaderAsync(IAsyncStreamReader<StreamingMessage> responseStream)
-        {
-            while (await responseStream.MoveNext())
-            {
-                await ProcessRequestAsync(responseStream.Current);
-            }
-        }
-
-        private Task ProcessRequestAsync(StreamingMessage request)
+        Task IMessageProcessor.ProcessMessageAsync(StreamingMessage message)
         {
             // Dispatch and return.
-            Task.Run(() => ProcessRequestCoreAsync(request));
+            _ = ProcessRequestCoreAsync(message);
```

Comment:
We should also update env reload case (line 116) to return capabilities and worker metadata. Can be a follow up PR I can take care of.

Reply:
Has the host been updated to support that? yes, we should, but either way, I'm inclined to say this should be a separate change